### PR TITLE
STYLE fixup cython linting

### PR DESCRIPTION
**Pull request by MarcoGorelli**
the alias means you can run `pre-commit run flake8-cython --all-files`

### CODE REVIEW

File: flake8/cython.cfg [hunk 1]
```diff
@@ -1,3 +1,17 @@
...
+extend_ignore=
+    # whitespace before '('
+    E211,
+    # missing whitespace around operator
+    E225,
+    # missing whitespace around arithmetic operator
+    E226,
+    # missing whitespace around bitwise or shift operator
+    E227,
+    # amiguous variable name (# FIXME maybe this one can be fixed)
```
Comment (jbrockmendel) [inline flake8/cython.cfg:12]: amiguous -> ambiguous

Comment (MarcoGorelli) [inline flake8/cython.cfg:12]:
```suggestion
    # ambiguous variable name (# FIXME maybe this one can be fixed)
```

**Comment by jreback** [comment]
can you merge master

**Review by mroeschke — APPROVED** [review]

File: pandas/_libs/lib.pyx [hunk 2]
```diff
@@ -2109,7 +2109,7 @@ cpdef bint is_interval_array(ndarray values):
                 if not (
                         util.is_float_object(val.left)
                         or util.is_integer_object(val.left)
```
Comment (jbrockmendel) [inline pandas/_libs/lib.pyx:2111]: i think dedent these?

File: pandas/_libs/parsers.pyx [hunk 3]
```diff
@@ -1439,8 +1439,8 @@ cdef _categorical_convert(parser_t *parser, int64_t col,
             COLITER_NEXT(it, word)
 
             if na_filter:
-                if kh_get_str_starts_item(na_hashset, word):
                 # is in NA values
+                if kh_get_str_starts_item(na_hashset, word):
```
Comment (jbrockmendel) [inline pandas/_libs/parsers.pyx:1443]: the linter wants comments before the `if`?

Comment (MarcoGorelli) [reply pandas/_libs/parsers.pyx:1444]: or just an indented comment - I've gone with that, thanks

**Comment by jreback** [comment]
thanks @MarcoGorelli very nice!

**Pull request merged**